mu/app.py

```diff
@@ -104,9 +104,18 @@ def run():
     # Setup the window.
     editor_window.closeEvent = editor.quit
     editor_window.setup(editor.debug_toggle_breakpoint, editor.theme)
-    # capture the filename passed by the os, if there was one
-    passed_filename = sys.argv[1] if len(sys.argv) > 1 else None
-    editor.restore_session(os.path.abspath(passed_filename))
+    # Restore the previous session
+    editor.restore_session(without_empty=len(sys.argv[1:]) > 0)
+    # capture the filename passed by the os, if there were any
+    for passed_filename in sys.argv[1:]:
+        try:
+            # abspath will fail for non-paths
+            editor.direct_load(os.path.abspath(passed_filename))
```

Comment:
hmmm.... I'm not sure about this approach. The code in `app.py` is essentially there to instantiate things and connect them together. This feels like logic that shouldn't be here (i.e. it should be within the editor, and all the editor needs are a bunch of paths for the passed in files). Does this make sense?

Reply:
Yep, I'll have a look at rearranging that